### PR TITLE
Adds node hooks: pre, mid and post update hook.

### PR DESCRIPTION
**Pull request by jtudelag**
mid hook after update node, but before making it schedulable again.

### CODE REVIEW

**Comment by jtudelag** [comment]
Followed the same approach used for master_hooks: https://github.com/openshift/openshift-ansible/commit/2c91510d4ce3baa92e8e054223659e4e3fa5dbcd

File: playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml [hunk 1]
```diff
@@ -42,12 +42,27 @@
...
+  - include_tasks: "{{ openshift_node_upgrade_pre_hook }}"
+    when: openshift_node_upgrade_pre_hook is defined
+
   post_tasks:
   - include_role:
       name: openshift_node
       tasks_from: upgrade.yml
     vars:
       openshift_node_upgrade_in_progress: True
+
```
Comment (juanluisvaladas) [inline playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml]: @jtudelag this line contains a trailing space which is causing the linter to fail

**Comment by mtnbikenc** [comment]
@jtudelag This approach looks good so far.  We are working on how to move forward with using hooks.  What is the specific problem you are trying to solve by implementing these hooks for the node upgrade?

**Comment by jtudelag** [comment]
@mtnbikenc The main use case is to update nodes O.S while updating OpenShift, so in the end, It will require only one reboot, and save time.

**Comment by sdodson** [comment]
@jtudelag Can you rebase this please? I think this is fine and aligns with what's already in place for masters.

**Comment by mtnbikenc** [comment]
Please rebase your branch to master and squash commits. Thanks

**Pull request closed without merging**